---
title: "Hands-on 2"
author: "`r authors`"
date: "`r Sys.Date()`"
output: bookdown::gitbook
site: bookdown::bookdown_site
---



# Hands-on 2: Key concepts and techniques

This session will let us familiarize with ecological concepts and
methods for analysing microbiome profiling data.


## Part I: Key concepts

Prepare a reproducible report and address the following questions.

 * **Alpha diversity** Estimate taxonomic diversity for each sample
     (function microbiome::alpha). Compare the results between two or
     more diversity indices (visually or statistically). See
     [microbiome tutorial](https://microbiome.github.io/tutorials/)
     for examples.

 * **Beta diversity** Visualize community variation with PCoA
     ([examples](https://microbiome.github.io/tutorials/Ordination.html)).
     Investigate the influence of the data transformations on
     statistical analysis: Visualize community variation with PCoA
     with the following options: 1) Bray-Curtis distances for
     compositional data; 2) Euclidean distances for CLR-transformed
     data.

 * **Prevalence** What is the most prevalent genus in the data (tip:
     microbiome::prevalence and sort)

 * **Core microbiota** Pick up the core microbiota including taxa that
     exceed 0.1% relative abundance in over 50% of the samples
     (prevalence). How many core taxa there are? You can read more about the core microbiota definition in [Salonen et al. 2012](https://www.clinicalmicrobiologyandinfection.com/article/S1198-743X(14)60962-9/fulltext). Visualize the core microbiota by following the available [examples](https://microbiome.github.io/tutorials/Core.html).


## Part II: Statistical testing

 * **Community-level comparisons** Use
     [PERMANOVA](https://microbiome.github.io/tutorials/PERMANOVA.html)
     to investigate whether the community composition differs between
     two groups of individuals (e.g. males and females, or some other
     grouping of your choice). You can also include covariates such as
     age and gender, and see how this affects the results.

 * **OTU-level comparisons** Use
     [DESeq2](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8)
     to identify which genera are associated with gender differences
     ([examples on
     DESeq2](https://microbiome.github.io/tutorials/all.html)). For
     more explanation on the method, see [DESeq2
     R/Bioconductor](http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html).

  * **Role of covariates** Note that including key covariates (diet,
     medication, age..) may have remarkable influence on data
     interpretation in the above analyses ([Falony et al. Science
     2016](https://science.sciencemag.org/content/352/6285/560.abstract?ijkey=ADV4ZnF4mHYIg&keytype=ref&siteid=sci))

 * Experiment with the other available tools in the [microbiome
   tutorial](https://microbiome.github.io/tutorials/)

 * Provide a compact summary of the data and analyses in the form of a
   reproducible Rmarkdown report. Include only selected summaries and
   analyses of your choice, and focus on delivering a clear and
   compact report.


 

 
