### Session 1 (9.30-11.30  / 2 hours)

**Overview of a standard bioinformatics workflow in microbiome research**

 * Ecosystem view of the gut microbiome: from bugs to communities and function
 * Preprocessing and quality control
  * Diagnostic plots and properties of microbiome data
  * Common methods: DADA2 (SVs) etc.
* Analysis and modeling of taxonomic composition
  * Principal coordinates analysis & other common visualization techniques 
  * Differential abundance analysis
  * Varieties of alpha and beta diversity
  * Identification of community types (DMM)
  * Quantification of associations with external factors such as age, diet, or medication
* Reproducible research and best practices in microbiome data science

### Session 2 (12-13 / 1 hour)

**Lecture: Modern statistics in human gut microbiome research**

 * Human gut microbiome: a quantitative view
 * Special properties of microbiome data 
 * The role of open data science and microbiome research


### Session 3 (14.30-17.30 / 3 hours including breaks)

 * Highlight how properties of microbiome data, such as sparsity,
   non-Gaussianity, and compositionality influences the choice and
   performance of the statistical methods: CLR; PCA vs. PCoA;
   community-level vs. subcommunity vs. species level; DESeq2; multiple
   testing;


**Hands-on session: introduction to R tools**

 * Part I: Raw data, preprocessing and quality control. Introduction
   to phyloseq & example data (from 16S rRNA amplicon profiling
   studies).
 
 * Part II: Key concepts: core, diversity, community types
 
 * Part III: Generating reproducible reports with Rmarkdown


 * Analysis and modeling of taxonomic composition
 * From targeted study designs to population monitoring
 * Exploratory data analysis
 * Examples on microbiome data visualization and diversity analysis
 * Differential abundance analysis
 * Quantification of associations with external factors such as age, diet, or medication

 microbiome, microbiomeutilities, phyloseq


### Session 4 (9.30-12:00 / 2.5 hours)

**Special properties of microbiome data**

 * 
In this session we will highlight how properties of microbiome data,
such as sparsity, non-Gaussianity, and compositionality influences the
choice and performance of the statistical methods: CLR; PCA vs. PCoA;
community-level vs. subcommunity vs. species level; DESeq2; multiple
testing;

 * Sparsity, compositionality, non-Gaussianity
 * The role of covariates: age, diet, or medication 
 * Influence on statistical analysis
 * Differential abundance analysis: DADA2
 * Visualization: PCoA
 * Co-occurrence networks: SparCC
 
# Misc

 * Analysis and modeling of taxonomic composition
 * From targeted study designs to population monitoring
 * Examples on microbiome data visualization and diversity analysis
