--- 
title: "Introduction to microbiome data science"
author: "Leo Lahti"
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
output: bookdown::gitbook
documentclass: book
bibliography: [book.bib, packages.bib]
biblio-style: apalike
link-citations: yes
github-repo: microbiome/test
description: "Course material"
---


# Introduction  

This course is organised in collaboration with [Department of Mathematics and Statistics University of Turku](https://www.utu.fi/en/units/sci/units/math/Pages/home.aspx), Finland. It will be held at [Radboud University](), Nijmegen on Sep 12-13, 2019.

The course will cover the standard workflow in taxonomic profiling studies, with a focus on human gut microbiota, ranging from low-level preprocessing to statistical analysis, visualization, and reproducible reporting.


## Preparing for the course

We recommend using your own laptop. If this is not possible, kindly contact the organizers.

Install the following software before the course:

* R [Download link](https://cran.r-project.org).    
* Rstudio [Download link](https://www.rstudio.com/products/rstudio/download/).  


## Useful functions/resources     

* [Base R](https://www.rstudio.com/wp-content/uploads/2016/10/r-cheat-sheet-3.pdf)  
* [R Markdown](https://www.rstudio.com/wp-content/uploads/2016/03/rmarkdown-cheatsheet-2.0.pdf)  
* [RStudio IDE](https://www.rstudio.com/wp-content/uploads/2016/01/rstudio-IDE-cheatsheet.pdf)  
* [ggplot2](https://www.rstudio.com/wp-content/uploads/2016/11/ggplot2-cheatsheet-2.1.pdf)  
* [phyloseq](http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0061217)  
* [microbiome](http://microbiome.github.io/microbiome/)  
* [R graphics cookbook](http://www.cookbook-r.com/Graphs/)  
* [List of R tools for microbiome analysis](https://microsud.github.io/Tools-Microbiome-Analysis/)  

## Focus    

The primary aim is introduce microbial community data analysis. There will be talks and discussion on theory and methodology for analysis of microbial community data. We will cover topics ranging from design of studies, sequencing technologies, importance of controls and standardized DNA processing. Supervised hands-on training covering analyses from raw reads using NG-Tax, downstream analysis in R for exploration and analysis of microbiome sequencing data will be priority. There will be a strong focus on using R, R Studio for moving towards reproducible and open science.   

## Target audience  
Anyone who plans to or is currently doing high throughput microbial community analysis.  

## Citation  

If you found this useful, please cite:  
Lahti Leo, Henrik Eckerman, Sudarshan Shetty (2019). Introduction to microbiome data science.


## License  

The 2-Clause BSD License


