# TODO

In a rough order of priority.

 * ~~Add the necessary dependencies to code/install_dependencies.R~~ **DONE - Tested  with R 3.6.1 on Mac OS Mojave and Windows 10**

 * ~~Check that all links work~~ **DONE**

 * ~~Add links to Radboud / course announcements / your own contact info
   for support / etc. where appropriate~~ 

 * ~~Consider changeing the example data set from atlas1006 to
   dietswap. The latter might be more similar to typical case studies
   encoutntered by the course participants.~~ **CHANGED TO DIETSWAP MAINLY BECAUSE MOST OF THEM WORK WITH 16S DATA** 

 * Add tips and links to useful examples and material and tips
   throughout to the handson exercise questions. This will help the
   students to find the necessary examples and code easily. Some of
   the R links in the study material might better go to the actual
   hands-on instructions?

 * Consider if you are willing to lead one of the hands-on sessions
   (with my help); and then we can switch for the other one?

 * ~~Prepare model solutions to the hands-on session questions; as
   reproducible scripts that can be run through as is. This will also
   help to test dependencies.~~ **See folder handson_code**

 * Update the literature and links in the study material section?

 * Check if there is something to add from the official course
   announcement (in the learning goals etc). **I think everything is covered that was mentioned in the announcement. One thing maybe: The course announcement mentions "...overall understanding of the available methods from preprocessing and quality control of the sequencing reads to downstream analysis and modeling of taxonomic composition within and between individuals or experimental groups." If the  preprocessing and quality control is covered in session 1, I think we cover everything we announced.**

 * Note that we can also modify the microbiome R package tutorials
   directly if necessary.
