---
title: "Example Hands on 2"
output: html_document
---

```{r, include=FALSE}
library(knitr)
opts_chunk$set(echo = TRUE, message = FALSE, warning = FALSE)
```

# Hands-on 2: Key concepts and techniques



## Part I: Key concepts

Before we can start with the exercises we need to load the required packages and the example data we work with:

```{r}
# load required packages
library(tidyverse)
library(microbiome)

#load data
data(dietswap) 
print(dietswap)

# rename the example data
pseq <- dietswap
# transform data for compositional analysis
pseq.rel <- microbiome::transform(pseq, transform = "compositional")
pseq.clr <- microbiome::transform(pseq, transform = "clr")
```




 * **Alpha diversity** Estimate taxonomic diversity for each sample
     (function microbiome::alpha). Compare the results between two or
     more diversity indices (visually or statistically). See
     [microbiome tutorial](https://microbiome.github.io/tutorials/)
     for examples.
     
```{r}
tab.alpha <- alpha(pseq)
head(tab.alpha) %>% kable()

# let us select 3 indeces and see how they correlate:
# I use the function ggpairs from the package GGally, which you
# might need to install first:

if(!require(GGally)) {
  install.packages("GGally")
}

tab.alpha %>%
  select(
    diversity_inverse_simpson, 
    diversity_gini_simpson, 
    diversity_shannon,
    chao1) %>%
  GGally::ggpairs()
```









 * **Beta diversity** Visualize community variation with PCoA
     ([examples](https://microbiome.github.io/tutorials/Ordination.html)).
     Investigate the influence of the data transformations on
     statistical analysis: Visualize community variation with PCoA
     with the following options:  
     1) Bray-Curtis distances for
     compositional data;  
     2) Euclidean distances for CLR-transformed
     data.
     
     
```{r}
# option 1: Bray Curtis
ord <- ordinate(pseq, method = "PCoA", distance = "bray")
plot_ordination(pseq, ord, color = "nationality")

# option 2: Euclidean distance for CLR transformed data
ord <- ordinate(pseq.clr, method = "PCoA", distance = "euclidean")
plot_ordination(pseq, ord, color = "nationality")
```


 * **Prevalence** What is the most prevalent genus in the data (tip:
     microbiome::prevalence and sort)

```{r}
```


```{r}
prevalence(pseq.rel, detection = 0, sort = TRUE) %>% 
  kable()
prevalence(pseq.rel, detection = 0.003, sort = TRUE) %>% 
  kable()
```


 * **Core microbiota** Pick up the core microbiota including taxa that
     exceed 0.1% relative abundance in over 50% of the samples
     (prevalence). How many core taxa there are? You can read more about the core microbiota definition in [Salonen et al. 2012](https://www.clinicalmicrobiologyandinfection.com/article/S1198-743X(14)60962-9/fulltext). Visualize the core microbiota by following the available [examples](https://microbiome.github.io/tutorials/Core.html).
     
     
```{r fig.height=12, fig.width=15}
pseq.core <- core(pseq.rel, detection = 0.001, prevalence = 0.5)
# how many taxa are there?
taxa(pseq.core) %>% length()
# alternative
core_members(pseq.rel, detection = 0.001, prevalence = 0.5) %>% length()

# vizualize core microbiota

# core line plots
detections <- c(0, 0.1, 0.5, 2, 5, 20)/100
prevalences <- seq(0.05, 1, 0.05)
plot_core(pseq.rel, prevalences = prevalences, detections = detections, plot.type = "lineplot")

# core heatmaps
prevalences <- seq(0.05, 1, 0.05)
detections <- 10^seq(log10(1e-3), log10(0.2), length = 10)
# define gray color palette
gray <- gray(seq(0, 1, length = 5))

p <- plot_core(
  pseq.rel, 
  plot.type = "heatmap", 
  colours = gray, 
  prevalences = prevalences, 
  detections = detections)
  

print(p)


# Core with absolute counts and horizontal view:
# and minimum population prevalence (given as percentage)
detections <- 10^seq(log10(1), log10(max(abundances(pseq))/100), length = 10)
library(RColorBrewer)
p <- plot_core(pseq, 
               plot.type = "heatmap",
               prevalences = prevalences,
               detections = detections,
               colours = rev(brewer.pal(5, "Spectral")),
               min.prevalence = .2, 
               horizontal = TRUE)
print(p)
```



## Part II: Statistical testing

 * **Community-level comparisons** Use
     [PERMANOVA](https://microbiome.github.io/tutorials/PERMANOVA.html)
     to investigate whether the community composition differs between
     two groups of individuals (e.g. males and females, or some other
     grouping of your choice). You can also include covariates such as
     age and gender, and see how this affects the results.

```{r}
# to perform PERMANOVA we need vegan
library(vegan)

# extract relative abundances and meta data
otu.relative <- abundances(pseq.rel)
pseq.meta <- meta(pseq.rel)

# Before we perform PERMANOVA we must test the assumption 
# that the multivariate spread between the groups is similar.
dist <- vegdist(t(otu.relative))
# in this specific case the assumption is not met for bmi_group
betadisper(dist, pseq.meta$bmi_group) %>%
  anova() %>% 
  kable()

# perform PERMANOVA
pm <- adonis(
  t(otu.relative) ~ bmi_group,
  data = pseq.meta,
  permutations = 999,
  method = "bray"
  )

# to obtain the effect size and the F-test p-value:
pm$aov.tab %>% as.data.frame() %>%
  kable()
```


```{r}
# to investigate the top factors
coef <- coefficients(pm)["bmi_group1", ]

# with base R
top.coef <- coef[rev(order(abs(coef)))[1:20]]
par(mar = c(3, 14, 2, 1))
barplot(sort(top.coef), horiz = T, las = 1, main = "Top taxa")

# with ggplot
coef %>% as.data.frame() %>% # convert to dataframe
  rownames_to_column("taxa") %>% # add the rownames as a column for plotting
  select(taxa, coef = ".") %>% # select and rename the columns
  arrange(desc(abs(coef))) %>% # sort by absolute value
  head(20) %>% # pick 20 highest coefficients
  arrange(desc(coef)) %>% # sort by value
  mutate(top_taxa = factor(taxa, levels = taxa)) %>% # for ggplot to keep order we need factor
  ggplot(aes(top_taxa, coef)) + # plot
  geom_bar(stat = "identity") +
  coord_flip()
```


Below I explore whether using a different distance metric as suggested by [Gloor et
     al. 2017](https://www.frontiersin.org/articles/10.3389/fmicb.2017.02224/full) makes a difference in the interpreations of the results:


```{r}
# extract relative abundances and meta data
otu.clr <- abundances(pseq.clr)
pseq.meta <- meta(pseq.clr)

# Before we perform PERMANOVA we must test the assumption 
# that the multivariate spread between the groups is similar.
dist.ait <- vegdist(t(otu.clr), method = "euclidean")
# in this specific case the assumption is not met for bmi_group
betadisper(dist.ait, pseq.meta$bmi_group) %>%
  anova() %>% 
  kable()

# perform PERMANOVA
pm.ait <- adonis(
  t(otu.clr) ~ bmi_group,
  data = pseq.meta,
  permutations = 999,
  method = "euclidean"
  )

# to obtain the effect size and the F-test p-value:
pm.ait$aov.tab %>% as.data.frame() %>%
  kable()

# to investigate the top factors
coef.ait <- coefficients(pm.ait)["bmi_group1", ]

# with base R
top.coef.ait <- coef.ait[rev(order(abs(coef.ait)))[1:20]]
par(mar = c(3, 14, 2, 1))
barplot(sort(top.coef.ait), horiz = T, las = 1, main = "Top taxa")

# with ggplot
coef.ait %>% as.data.frame() %>% # convert to dataframe
  rownames_to_column("taxa") %>% # add the rownames as a column for plotting
  select(taxa, coef = ".") %>% # select and rename the columns
  arrange(desc(abs(coef.ait))) %>% # sort by absolute value
  head(20) %>% # pick 20 highest coefficients
  arrange(desc(coef)) %>% # sort by value
  mutate(top_taxa = factor(taxa, levels = taxa)) %>% # for ggplot to keep order we need factor
  ggplot(aes(top_taxa, coef)) + # plot
  geom_bar(stat = "identity") +
  coord_flip()
```

We see that in both cases the conclusions would be similar: The assumptions of similar multivariate spread between groups is violated. The PERMANOVA would suggest that belonging to a different BMI group explains a small portion of variation in the microbiota composition. When it comes to most important genera, there is a clear overlap. However, when using Aitchison distance the differences in importance are less extreme.




 * **OTU-level comparisons** Use
     [DESeq2](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8)
     to identify which genera are associated with gender differences
     ([examples on
     DESeq2](https://microbiome.github.io/tutorials/all.html)). For
     more explanation on the method, see [DESeq2
     R/Bioconductor](http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html).
     
```{r}
library(DESeq2)
# start by converting phyloseq object to deseq2 format
ds2 <- phyloseq_to_deseq2(pseq, ~nationality)
# Run DeSeq2 analysis (all taxa at once!)
dds <- DESeq(ds2)
```

Compare nationalities based on DESeq2 results. Note that covariates are ignored.

```{r nationality}
res1 <- results(dds, contrast = c("nationality", "AAM", "AFR"))
kable(head(as.data.frame(res1)))
```


MA-plot

```{r maplot}
plotMA(res1, cex = 1)
```

Summarize the largest differences as a table:

```{r}
res <- results(dds)
df <- as.data.frame(res) %>% 
  rownames_to_column("taxon") %>%
  filter(padj < 0.05 & abs(log2FoldChange) > 1.5) %>%  
  arrange(padj, desc(abs(log2FoldChange))) %>%
  mutate_if(is.numeric, round, 3) %>% # round to 3 decimal places
  kable()
df
```


  * **Role of covariates** Note that including key covariates (diet,
     medication, age..) may have remarkable influence on data
     interpretation in the above analyses ([Falony et al. Science
     2016](https://science.sciencemag.org/content/352/6285/560.abstract?ijkey=ADV4ZnF4mHYIg&keytype=ref&siteid=sci))

 * Experiment with the other available tools in the [microbiome
   tutorial](https://microbiome.github.io/tutorials/)
   

   
   

 * Provide a compact summary of the data and analyses in the form of a
   reproducible Rmarkdown report. Include only selected summaries and
   analyses of your choice, and focus on delivering a clear and
   compact report.
