<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<meta property="og:title" content="3.1 OTU or ASVs or sOTUs | OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018 v2.0" />
<meta property="og:type" content="book" />





<meta name="author" content="Sudarshan A. Shetty, Leo Lahti, Gerben DA. Hermes" />

<meta name="date" content="2019-07-29" />


<meta name="description" content="3.1 OTU or ASVs or sOTUs | OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018 v2.0">

<title>3.1 OTU or ASVs or sOTUs | OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018 v2.0</title>

<script src="libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="libs/navigation-1.1/tabsets.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
</style>
</head>

<body>

<div class="container-fluid main-container">


<div class="row">
<div class="col-sm-12">
<div id="TOC">
<ul>
<li class="has-sub"><a href="1-introduction.html#introduction"><span class="toc-section-number">1</span> Introduction</a><ul>
<li><a href="1-1-preparing-for-the-course.html#preparing-for-the-course"><span class="toc-section-number">1.1</span> Preparing for the course</a></li>
<li><a href="1-2-useful-functionsresources.html#useful-functionsresources"><span class="toc-section-number">1.2</span> Useful functions/resources</a></li>
<li><a href="1-3-focus.html#focus"><span class="toc-section-number">1.3</span> Focus</a></li>
<li><a href="1-4-target-audience.html#target-audience"><span class="toc-section-number">1.4</span> Target audience</a></li>
</ul></li>
<li class="has-sub"><a href="2-citation.html#citation"><span class="toc-section-number">2</span> Citation</a><ul>
<li><a href="2-1-license.html#license"><span class="toc-section-number">2.1</span> License</a></li>
</ul></li>
<li class="has-sub"><a href="3-set-up-and-pre-processing.html#set-up-and-pre-processing"><span class="toc-section-number">3</span> Set-up and Pre-processing</a><ul>
<li><a href="3-1-otu-or-asvs-or-sotus.html#otu-or-asvs-or-sotus"><span class="toc-section-number">3.1</span> OTU or ASVs or sOTUs</a></li>
<li><a href="3-2-general-overview.html#general-overview"><span class="toc-section-number">3.2</span> General overview</a></li>
<li><a href="3-3-structure.html#structure"><span class="toc-section-number">3.3</span> Structure</a></li>
<li><a href="3-4-making-a-phyloseq-object.html#making-a-phyloseq-object"><span class="toc-section-number">3.4</span> Making a phyloseq object</a></li>
</ul></li>
</ul>
</div>
</div>
</div>
<div class="row">
<div class="col-sm-12">
<div id="otu-or-asvs-or-sotus" class="section level2">
<h2><span class="header-section-number">3.1</span> OTU or ASVs or sOTUs</h2>
<p>For past few years (maybe decade), identifying Operational taxonomic units (OTUs) from raw sequences used clustering approach. Using 97% identity cut-off was a standard approach and often closed reference OTU picking was accepted in the sicentific community. During the time of the development of tools and bioinformatics approaches this was possibly the best available method. However, as with many other fields of science, the knowledge has been updated. Evolution of bioinformatics approaches is a constant process. Based on current knowledge, the cons of 97% OTU picking stratergy (using clustering approaches) have out-weighed the pros (eg. less time).</p>
<p>Recent approaches are now focused towards Amplicon Seuence Variants/sOTUs:<br />
* <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3864673/">Oligotyping</a><br />
* <a href="http://dx.doi.org/10.1128/mSystems.00191-16">Deblur</a><br />
* <a href="https://www.nature.com/articles/nmeth.3869">DADA2</a><br />
* <a href="https://f1000research.com/articles/5-1791/v1">NG-Tax</a></p>
<p>All above approaches have one common theme, they avoid 97% clustering and focus on minor differences (in many cases single nucleotide variations) to identify unique ASVs/sOTU.</p>
<p>Note: Some how naming is different and variable. For this purpose and in this book, I will stick to ASVs when data from NG-tax is used.</p>
<p>In this, section, we will compare outputs from 97% OTU picking approach and NG-tax approach.<br />
The data used here is the 16S rRNA gene variable region (V1-V3) for 97% OTU-pciking. The raw reads were processed using QIIME 1.9.1, SortMeRNA, and OTU picking was done using the closed-reference OTU-picking at 97% identity.</p>
<p>For NG-Tax, we use the same raw data and processed through default settings.</p>
<p>Here, we do not aim to bench mark. For this course, we aim to show differences between results from two approaches.</p>
<p>For down stream analysis of *.biom files we use <a href="https://joey711.github.io/phyloseq/">Phyloseq</a> and <a href="http://microbiome.github.io/microbiome/">microbiome</a>.<br />
Kindly cite all the packages and tools that were used in your analysis as listed at the end of each document in <code>sessionInfo</code>. Also make sure that you provide the workflow and scripts you used for analysis atleast as supplementary material with your research article.<br />
Check <a href="http://www.statmethods.net/">Quick-R</a>.</p>
</div>
<p style="text-align: center;">
<a href="3-set-up-and-pre-processing.html"><button class="btn btn-default">Previous</button></a>
<a href="3-2-general-overview.html"><button class="btn btn-default">Next</button></a>
</p>
</div>
</div>


</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

</body>
</html>
