<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>3 Course overview | Introduction to microbiome data science</title>
  <meta name="description" content="Course material" />
  <meta name="generator" content="bookdown 0.12 and GitBook 2.6.7" />

  <meta property="og:title" content="3 Course overview | Introduction to microbiome data science" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Course material" />
  <meta name="github-repo" content="microbiome/test" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="3 Course overview | Introduction to microbiome data science" />
  
  <meta name="twitter:description" content="Course material" />
  

<meta name="author" content="Leo Lahti" />


<meta name="date" content="2019-07-29" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="set-up-and-pre-processing.html">
<link rel="next" href="differential-abundance-testing-for-univariate-data.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#preparing-for-the-course"><i class="fa fa-check"></i><b>1.1</b> Preparing for the course</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#useful-functionsresources"><i class="fa fa-check"></i><b>1.2</b> Useful functions/resources</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#focus"><i class="fa fa-check"></i><b>1.3</b> Focus</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#target-audience"><i class="fa fa-check"></i><b>1.4</b> Target audience</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#citation"><i class="fa fa-check"></i><b>1.5</b> Citation</a></li>
<li class="chapter" data-level="1.6" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i><b>1.6</b> License</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html"><i class="fa fa-check"></i><b>2</b> Set-up and Pre-processing</a><ul>
<li class="chapter" data-level="2.1" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#otu-or-asvs-or-sotus"><i class="fa fa-check"></i><b>2.1</b> OTU or ASVs or sOTUs</a></li>
<li class="chapter" data-level="2.2" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#general-overview"><i class="fa fa-check"></i><b>2.2</b> General overview</a></li>
<li class="chapter" data-level="2.3" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#structure"><i class="fa fa-check"></i><b>2.3</b> Structure</a></li>
<li class="chapter" data-level="2.4" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#making-a-phyloseq-object"><i class="fa fa-check"></i><b>2.4</b> Making a phyloseq object</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="course-overview.html"><a href="course-overview.html"><i class="fa fa-check"></i><b>3</b> Course overview</a><ul>
<li class="chapter" data-level="3.1" data-path="course-overview.html"><a href="course-overview.html#september"><i class="fa fa-check"></i><b>3.1</b> 12 September:</a></li>
<li class="chapter" data-level="3.2" data-path="course-overview.html"><a href="course-overview.html#september-1"><i class="fa fa-check"></i><b>3.2</b> 13 September:</a></li>
<li class="chapter" data-level="3.3" data-path="course-overview.html"><a href="course-overview.html#material"><i class="fa fa-check"></i><b>3.3</b> Material</a></li>
<li class="chapter" data-level="3.4" data-path="course-overview.html"><a href="course-overview.html#literature"><i class="fa fa-check"></i><b>3.4</b> Literature</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html"><i class="fa fa-check"></i><b>4</b> Differential abundance testing for univariate data</a><ul>
<li class="chapter" data-level="4.1" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#load-example-data"><i class="fa fa-check"></i><b>4.1</b> Load example data</a></li>
<li class="chapter" data-level="4.2" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#visual-comparison-of-two-groups"><i class="fa fa-check"></i><b>4.2</b> Visual comparison of two groups</a></li>
<li class="chapter" data-level="4.3" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#statistical-comparison-of-two-groups"><i class="fa fa-check"></i><b>4.3</b> Statistical comparison of two groups</a></li>
<li class="chapter" data-level="4.4" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#investigate-assumptions-of-the-t-test"><i class="fa fa-check"></i><b>4.4</b> Investigate assumptions of the t-test</a></li>
<li class="chapter" data-level="4.5" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#compare-results-between-parametric-and-non-parametric-tests"><i class="fa fa-check"></i><b>4.5</b> Compare results between parametric and non-parametric tests</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html"><i class="fa fa-check"></i><b>5</b> Linear models: the role of covariates</a><ul>
<li class="chapter" data-level="5.1" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html#fitting-a-linear-model"><i class="fa fa-check"></i><b>5.1</b> Fitting a linear model</a></li>
<li class="chapter" data-level="5.2" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html#interpreting-linear-model-output"><i class="fa fa-check"></i><b>5.2</b> Interpreting linear model output</a></li>
<li class="chapter" data-level="5.3" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html#covariate-testing"><i class="fa fa-check"></i><b>5.3</b> Covariate testing</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html"><i class="fa fa-check"></i><b>6</b> Advanced models for differential abundance</a><ul>
<li class="chapter" data-level="6.1" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#particular-properties-of-taxonomic-profiling-data"><i class="fa fa-check"></i><b>6.1</b> Particular properties of taxonomic profiling data</a><ul>
<li class="chapter" data-level="6.1.1" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#discrete-count-data"><i class="fa fa-check"></i><b>6.1.1</b> Discrete count data</a></li>
<li class="chapter" data-level="6.1.2" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#sparsity"><i class="fa fa-check"></i><b>6.1.2</b> Sparsity</a></li>
<li class="chapter" data-level="6.1.3" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#rarity"><i class="fa fa-check"></i><b>6.1.3</b> Rarity</a></li>
<li class="chapter" data-level="6.1.4" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#overdispersion"><i class="fa fa-check"></i><b>6.1.4</b> Overdispersion</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#generalized-linear-models-a-brief-overview"><i class="fa fa-check"></i><b>6.2</b> Generalized linear models: a brief overview</a></li>
<li class="chapter" data-level="6.3" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#deseq2-differential-abundance-testing-for-sequencing-data"><i class="fa fa-check"></i><b>6.3</b> DESeq2: differential abundance testing for sequencing data</a><ul>
<li class="chapter" data-level="6.3.1" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#fitting-deseq2"><i class="fa fa-check"></i><b>6.3.1</b> Fitting DESeq2</a></li>
<li class="chapter" data-level="6.3.2" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#comparison-between-deseq2-and-standard-models"><i class="fa fa-check"></i><b>6.3.2</b> Comparison between DESeq2 and standard models</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="multivariate-comparisons-of-microbial-community-composition.html"><a href="multivariate-comparisons-of-microbial-community-composition.html"><i class="fa fa-check"></i><b>7</b> Multivariate comparisons of microbial community composition</a><ul>
<li class="chapter" data-level="7.1" data-path="multivariate-comparisons-of-microbial-community-composition.html"><a href="multivariate-comparisons-of-microbial-community-composition.html#permanova"><i class="fa fa-check"></i><b>7.1</b> PERMANOVA</a></li>
<li class="chapter" data-level="7.2" data-path="multivariate-comparisons-of-microbial-community-composition.html"><a href="multivariate-comparisons-of-microbial-community-composition.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>7.2</b> Checking the homogeneity condition</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="further-exercises.html"><a href="further-exercises.html"><i class="fa fa-check"></i><b>8</b> Further exercises</a><ul>
<li class="chapter" data-level="8.1" data-path="further-exercises.html"><a href="further-exercises.html#compositionality"><i class="fa fa-check"></i><b>8.1</b> Compositionality</a></li>
<li class="chapter" data-level="8.2" data-path="further-exercises.html"><a href="further-exercises.html#redundancy-analysis-rda"><i class="fa fa-check"></i><b>8.2</b> Redundancy analysis (RDA)</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="citation-1.html"><a href="citation-1.html"><i class="fa fa-check"></i><b>9</b> Citation</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Introduction to microbiome data science</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="course-overview" class="section level1">
<h1><span class="header-section-number">3</span> Course overview</h1>
<p><strong>Data exploration and analysis in gut microbiome profiling studies</strong></p>
<div id="september" class="section level2">
<h2><span class="header-section-number">3.1</span> 12 September:</h2>
<p>Session 1 (9.30-11.30 / 2 hours) Overview of a standard bioinformatics workflow in microbiome research. - Ecosystem view of the gut microbiome: from bugs to communities and function - Preprocessing and quality control - Diagnostic plots and properties of microbiome data - Common methods: DADA2 (SVs) etc. - Analysis and modeling of taxonomic composition - Principal coordinates analysis &amp; other common visualization techniques - Differential abundance analysis - Varieties of alpha and beta diversity - Identification of community types (DMM) - Quantification of associations with external factors such as age, diet, or medication - Reproducible research and best practices in microbiome data science</p>
<p>Session 2 (12-13 / 1 hour) KEYNOTE: Modern statistics in human gut microbiome research</p>
<ul>
<li><p>Overview of a standard bioinformatics workflow in microbiome research.</p></li>
<li><p>Highlight how properties of microbiome data, such as sparsity, non-Gaussianity, and compositionality influences the choice and performance of the statistical methods: CLR; PCA vs. PCoA; community-level vs. subcommunity vs. species level; DESeq2; multiple testing;</p></li>
<li><p>Role of reproducible research and best practices in microbiome data science</p></li>
</ul>
<p>Session 3 (14.30-17.30 / 3 hours including breaks)</p>
<p>Hands-on session: introduction to R tools (microbiome, microbiomeutilities, phyloseq, other)</p>
<p>3A) Introduction to phyloseq &amp; example data (from 16S rRNA amplicon profiling studies)</p>
<p>3B) Generating a reproducible report with Rmarkdown</p>
<p>3C) Examples on microbiome data visualization and diversity analysis</p>
</div>
<div id="september-1" class="section level2">
<h2><span class="header-section-number">3.2</span> 13 September:</h2>
<p>Session 4 (9.30-12:00 / 2.5 hours)</p>
<p>Lunch 12:00-13:30 (1.5 hours)</p>
<p>Session 5 (13:30-15:00 / 1.5 hours)</p>
<p>Downstream analysis and modeling of taxonomic composition within and between individuals or experimental groups</p>
<p>Break (15:00-15:30 / 0.5 hours)</p>
<p>Session 6 (15:30-17:00 / 1.5 hours) - Pointers to other tools &amp; environments (QIIME2, Anvi’o) - Wrap-up &amp; conclusions</p>
</div>
<div id="material" class="section level2">
<h2><span class="header-section-number">3.3</span> Material</h2>
<p><a href="https://datacarpentry.org/R-ecology-lesson/" class="uri">https://datacarpentry.org/R-ecology-lesson/</a></p>
</div>
<div id="literature" class="section level2">
<h2><span class="header-section-number">3.4</span> Literature</h2>
<ul>
<li><p>Modern Statistics for Modern Biology. Holmes &amp; Huber (2018):. URL: <a href="http://web.stanford.edu/class/bios221/book/" class="uri">http://web.stanford.edu/class/bios221/book/</a></p></li>
<li><p>Signatures of ecological processes in microbial community time series. Faust K et al. Microbiome, 6(120) 2018</p></li>
<li><p>Multi-stability and the origin of microbial community types. Faust et al. ISME Journal, 11:2159–2166, 2017.</p></li>
<li><p>Linking statistical and ecological theory: Hubbell’s unified neutral theory of biodiversity as a hierarchical Dirichlet process. Harris et al. Proceedings of the IEEE, 105(3):516–529, 2017.</p></li>
<li><p>Metagenomics meets time series analysis: unraveling microbial community dynamics. Faust K. Current Opinion in Microbiology, 25:56–66, 2015.</p></li>
<li><p>Microbial communities as dynamical systems. Gonze D et al. Current Opinion in Microbiology, 44:41–49, 2018.</p></li>
<li><p>Microbiome Data Science. Shetty &amp; Lahti, 2019. URL: <a href="https://openresearchlabs.github.io/publications/papers/2018-Shetty-Lahti-MDS.pdf" class="uri">https://openresearchlabs.github.io/publications/papers/2018-Shetty-Lahti-MDS.pdf</a></p></li>
</ul>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="set-up-and-pre-processing.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="differential-abundance-testing-for-univariate-data.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
