<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>3 Hands-on 1: introduction to R tools | Introduction to microbiome data science</title>
  <meta name="description" content="Course material" />
  <meta name="generator" content="bookdown 0.12 and GitBook 2.6.7" />

  <meta property="og:title" content="3 Hands-on 1: introduction to R tools | Introduction to microbiome data science" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Course material" />
  <meta name="github-repo" content="microbiome/course_nijmegen2019" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="3 Hands-on 1: introduction to R tools | Introduction to microbiome data science" />
  
  <meta name="twitter:description" content="Course material" />
  

<meta name="author" content="Leo Lahti and Henrik Eckermann" />


<meta name="date" content="2019-08-12" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="schedule.html">
<link rel="next" href="hands-on-2-key-concepts-and-techniques.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />










</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Overview</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#preparing-for-the-course"><i class="fa fa-check"></i><b>1.1</b> Preparing for the course</a><ul>
<li class="chapter" data-level="1.1.1" data-path="index.html"><a href="index.html#software-setup"><i class="fa fa-check"></i><b>1.1.1</b> Software setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="schedule.html"><a href="schedule.html"><i class="fa fa-check"></i><b>2</b> Schedule</a><ul>
<li class="chapter" data-level="2.1" data-path="schedule.html"><a href="schedule.html#september-12-thu"><i class="fa fa-check"></i><b>2.1</b> September 12 (Thu)</a></li>
<li class="chapter" data-level="2.2" data-path="schedule.html"><a href="schedule.html#september-13-fri"><i class="fa fa-check"></i><b>2.2</b> September 13 (Fri)</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="hands-on-1-introduction-to-r-tools.html"><a href="hands-on-1-introduction-to-r-tools.html"><i class="fa fa-check"></i><b>3</b> Hands-on 1: introduction to R tools</a><ul>
<li class="chapter" data-level="3.1" data-path="hands-on-1-introduction-to-r-tools.html"><a href="hands-on-1-introduction-to-r-tools.html#part-i-reproducible-reporting-with-rmarkdown"><i class="fa fa-check"></i><b>3.1</b> Part I: Reproducible reporting with Rmarkdown</a></li>
<li class="chapter" data-level="3.2" data-path="hands-on-1-introduction-to-r-tools.html"><a href="hands-on-1-introduction-to-r-tools.html#part-ii-phyloseq-data-structure"><i class="fa fa-check"></i><b>3.2</b> Part II: Phyloseq data structure</a></li>
<li class="chapter" data-level="3.3" data-path="hands-on-1-introduction-to-r-tools.html"><a href="hands-on-1-introduction-to-r-tools.html#part-iii-data-manipulation-and-exploration"><i class="fa fa-check"></i><b>3.3</b> Part III: Data manipulation and exploration</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="hands-on-2-key-concepts-and-techniques.html"><a href="hands-on-2-key-concepts-and-techniques.html"><i class="fa fa-check"></i><b>4</b> Hands-on 2: Key concepts and techniques</a><ul>
<li class="chapter" data-level="4.1" data-path="hands-on-2-key-concepts-and-techniques.html"><a href="hands-on-2-key-concepts-and-techniques.html#part-i-key-concepts"><i class="fa fa-check"></i><b>4.1</b> Part I: Key concepts</a></li>
<li class="chapter" data-level="4.2" data-path="hands-on-2-key-concepts-and-techniques.html"><a href="hands-on-2-key-concepts-and-techniques.html#part-ii-statistical-testing"><i class="fa fa-check"></i><b>4.2</b> Part II: Statistical testing</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="study-material.html"><a href="study-material.html"><i class="fa fa-check"></i><b>5</b> Study material</a><ul>
<li class="chapter" data-level="5.1" data-path="study-material.html"><a href="study-material.html#lecture-slides"><i class="fa fa-check"></i><b>5.1</b> Lecture slides</a></li>
<li class="chapter" data-level="5.2" data-path="study-material.html"><a href="study-material.html#r-programming-additional-resources"><i class="fa fa-check"></i><b>5.2</b> R programming: additional resources</a></li>
<li class="chapter" data-level="5.3" data-path="study-material.html"><a href="study-material.html#further-reading"><i class="fa fa-check"></i><b>5.3</b> Further reading</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Introduction to microbiome data science</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="hands-on-1-introduction-to-r-tools" class="section level1">
<h1><span class="header-section-number">3</span> Hands-on 1: introduction to R tools</h1>
<p>The hands-on sessions consists of a set of questions and example data. Solve the exercises by taking advantage of the online examples and resources that are pointed out in the study material. There is often more than one way to solve a given task. It is assumed that you have already installed the <a href="https://microbiome.github.io/course_nijmegen2019/index.html#preparing-for-the-course">required software</a>. Do not hesitate to ask support from the course assistants.</p>
<div id="part-i-reproducible-reporting-with-rmarkdown" class="section level2">
<h2><span class="header-section-number">3.1</span> Part I: Reproducible reporting with Rmarkdown</h2>
<p>Reproducible reporting is the starting point for reporting. Perform the following tasks:</p>
<ul>
<li><p>Create a Rmarkdown template in RStudio, and render it into a
document (markdown, PDF, docx or other format)</p></li>
<li><p>Experiment with different options with <a href="https://www.rstudio.com/wp-content/uploads/2016/03/rmarkdown-cheatsheet-2.0.pdf">Rmarkdown</a></p></li>
</ul>
</div>
<div id="part-ii-phyloseq-data-structure" class="section level2">
<h2><span class="header-section-number">3.2</span> Part II: Phyloseq data structure</h2>
<p>Next, let us investigate in more detail how taxonomic profiling data is organized in R. The data structure is called phyloseq (<a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0061217">McMurdie and Holmes, 2013</a>).</p>
<p>In this course, we focus on downstream analysis of taxonomic profiling data, and assume that the data has already been appropriately preprocessed and converted in the phyloseq format. The exact steps vary based on the 16S profiling platform. Hands-on examples for converting the Illumina MiSeq sequencing data into R phyloseq object can be found from Bioconductor Workflow for Microbiome Data Analysis: from raw reads to community analyses (<a href="https://f1000research.com/articles/5-1492/v2">Callahan et al. F1000, 2016</a>).</p>
<p>Provide the answers to at least some of the following questions in the form of a reproducible report (with Rmarkdown). Include the code that you use to retrieve the results. We will release code that provides solutions to all exercises.</p>
<ul>
<li><p><strong>Load example data</strong> of the gut microbiome profiling of 1006
western adults from <a href="https://microbiome.github.io/tutorials/Data.html">microbiome R
package</a>. The
example data sets are already in the <a href="http://joey711.github.io/phyloseq/import-data">phyloseq
format</a>. R
provides many readily available tools to handle and analyze
taxonomic profiling data that is provided in this format.</p></li>
<li><p><strong>Microbal abundances / OTU Table</strong> retrieve the microbial (OTU)
abundance table from the example data set (phyloseq object). This
example data set has abundances at the genus level instead of OTU
level. The taxonomic groupings in this 16S data set are only
approximations to exact genus-level taxonomic classifications. Tip:
microbiome::abundances or phyloseq::otu_table</p></li>
<li><p><strong>OTU tables</strong> Retrieve the OTU table and investigate: 1) How many
different samples and genus-level groups this phyloseq object has?
What is the maximum abundance of Akkermansia in this data set?
Tips: see data operations in the <a href="https://microbiome.github.io/tutorials/">microbiome
tutorial</a></p></li>
<li><p>Draw a histogram of library sizes (total number of reads per
sample). You can use the readcount function, or count the sum of
reads per sample by using the colSums command applied on the otu
table abundances. The example data set is from phylogenetic
microarray, hence the numbers correspond to light intensity on the
array rather than sequencing reads but the interpretation is
analogous and we can ignore this detail for demonstration
purposes. The key idea is that different samples may have different
bacterial load. Check <a href="https://www.nature.com/articles/nature24460">Vandeputte et
al. 2017</a> for further
discussion on the differences between absolute and relative
quantification of microbial abundances.</p></li>
<li><p><strong>Taxonomy table</strong> Retrieve the taxonomy table and print out the
first few lines of it with the R command head(). Investigate how
many different phylum-level groups this phyloseq object has? Tips:
phyloseq::tax_table and unique.</p></li>
<li><p><strong>Sample metadata</strong> Retrieve sample metadata. How many unique
subjects this data set has? What is the male/femal ratio? Draw a
histogram of subject ages. Tips: microbiome::meta</p></li>
</ul>
</div>
<div id="part-iii-data-manipulation-and-exploration" class="section level2">
<h2><span class="header-section-number">3.3</span> Part III: Data manipulation and exploration</h2>
<ul>
<li><p><strong>Subsetting</strong> Pick a subset of the phyloseq object including only
obese individuals who are over 40 years old. How many there are?
Tips: subset_samples</p></li>
<li><p><strong>Transformations</strong> The phyloseq data has absolute abundances. Use
the microbiome::transform command to convert these into relative
abundances. Compare Akkermansia abundance using the example data
before and after the compositionality transformation (with a
cross-plot, for instance). Finally, compare the results to CLR-
transformed data (see e.g. <a href="https://www.frontiersin.org/articles/10.3389/fmicb.2017.02224/full">Gloor et
al. 2017</a>)</p></li>
<li><p><strong>Visual exploration</strong> Visualize the population distribution of
abundances for the following taxa: Akkermansia, Dialister,
Gemella. Do the same for CLR-transformed abundances. Tip:
microbiome::plot_density / microbiome::transform</p></li>
<li><p>Experiment with other <a href="https://microbiome.github.io/tutorials/Preprocessing.html">phyloseq data manipulation
tools</a></p></li>
</ul>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="schedule.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="hands-on-2-key-concepts-and-techniques.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
