<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Introduction to microbiome data science</title>
  <meta name="description" content="Course material" />
  <meta name="generator" content="bookdown 0.12 and GitBook 2.6.7" />

  <meta property="og:title" content="Introduction to microbiome data science" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Course material" />
  <meta name="github-repo" content="microbiome/course_nijmegen2019" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Introduction to microbiome data science" />
  
  <meta name="twitter:description" content="Course material" />
  

<meta name="author" content="Leo Lahti and Henrik Eckermann" />


<meta name="date" content="2019-08-21" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  

<link rel="next" href="schedule.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />










</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Overview</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#preparing-for-the-course"><i class="fa fa-check"></i><b>1.1</b> Preparing for the course</a><ul>
<li class="chapter" data-level="1.1.1" data-path="index.html"><a href="index.html#software-setup"><i class="fa fa-check"></i><b>1.1.1</b> Software setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="schedule.html"><a href="schedule.html"><i class="fa fa-check"></i><b>2</b> Schedule</a><ul>
<li class="chapter" data-level="2.1" data-path="schedule.html"><a href="schedule.html#september-12-thu"><i class="fa fa-check"></i><b>2.1</b> September 12 (Thu)</a></li>
<li class="chapter" data-level="2.2" data-path="schedule.html"><a href="schedule.html#september-13-fri"><i class="fa fa-check"></i><b>2.2</b> September 13 (Fri)</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="hands-on-1-introduction-to-r-tools.html"><a href="hands-on-1-introduction-to-r-tools.html"><i class="fa fa-check"></i><b>3</b> Hands-on 1: introduction to R tools</a><ul>
<li class="chapter" data-level="3.1" data-path="hands-on-1-introduction-to-r-tools.html"><a href="hands-on-1-introduction-to-r-tools.html#part-i-reproducible-reporting-with-rmarkdown"><i class="fa fa-check"></i><b>3.1</b> Part I: Reproducible reporting with Rmarkdown</a></li>
<li class="chapter" data-level="3.2" data-path="hands-on-1-introduction-to-r-tools.html"><a href="hands-on-1-introduction-to-r-tools.html#part-ii-phyloseq-data-structure"><i class="fa fa-check"></i><b>3.2</b> Part II: Phyloseq data structure</a></li>
<li class="chapter" data-level="3.3" data-path="hands-on-1-introduction-to-r-tools.html"><a href="hands-on-1-introduction-to-r-tools.html#part-iii-data-manipulation-and-exploration"><i class="fa fa-check"></i><b>3.3</b> Part III: Data manipulation and exploration</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="hands-on-2-key-concepts-and-techniques.html"><a href="hands-on-2-key-concepts-and-techniques.html"><i class="fa fa-check"></i><b>4</b> Hands-on 2: Key concepts and techniques</a><ul>
<li class="chapter" data-level="4.1" data-path="hands-on-2-key-concepts-and-techniques.html"><a href="hands-on-2-key-concepts-and-techniques.html#part-i-key-concepts"><i class="fa fa-check"></i><b>4.1</b> Part I: Key concepts</a></li>
<li class="chapter" data-level="4.2" data-path="hands-on-2-key-concepts-and-techniques.html"><a href="hands-on-2-key-concepts-and-techniques.html#part-ii-statistical-testing"><i class="fa fa-check"></i><b>4.2</b> Part II: Statistical testing</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="study-material.html"><a href="study-material.html"><i class="fa fa-check"></i><b>5</b> Study material</a><ul>
<li class="chapter" data-level="5.1" data-path="study-material.html"><a href="study-material.html#lecture-slides"><i class="fa fa-check"></i><b>5.1</b> Lecture slides</a></li>
<li class="chapter" data-level="5.2" data-path="study-material.html"><a href="study-material.html#r-programming-additional-resources"><i class="fa fa-check"></i><b>5.2</b> R programming: additional resources</a></li>
<li class="chapter" data-level="5.3" data-path="study-material.html"><a href="study-material.html#further-reading"><i class="fa fa-check"></i><b>5.3</b> Further reading</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Introduction to microbiome data science</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">Introduction to microbiome data science</h1>
<p class="author"><em>Leo Lahti and Henrik Eckermann</em></p>
<p class="date"><em>2019-08-21</em></p>
</div>
<div id="overview" class="section level1">
<h1><span class="header-section-number">1</span> Overview</h1>
<p>This course provides an overview of the standard bioinformatics workflow in taxonomic profiling studies, ranging from data preprocessing to statistical analysis and reproducible reporting, with a focus on examples from human gut microbiota studies.</p>
<p><strong>Target audience</strong> Advanced students and applied researchers who wish to develop their skills in microbial community analysis.</p>
<p><strong>Venue</strong> <a href="">Radboud University</a>, Nijmegen. Sep 12-13, 2019, organised in collaboration with <a href="https://www.utu.fi/en/university/faculty-of-science-and-engineering/department-of-future-technologies">University of Turku</a>, Finland.</p>
<p><strong>Learning goals</strong> The students will become familiar with standard bioinformatics concepts and methods in taxonomic profiling studies of the human microbiome. This includes better understanding of the specific statistical challenges, practical hands-on experience with the commonly used methods, and reproducible research with R. After the course the students know how to approach new programming tasks by utilizing available documentation and R tools.</p>
<p><strong>Reusing the material</strong> Unless otherwise stated, the teaching materials can be freely used, modified and shared based on the open 2-Clause BSD License. Kindly cite the material as follows: “Lahti Leo, Henrik Eckerman, Sudarshan Shetty (2019). Introduction to microbiome data science. URL: <a href="https://microbiome.github.io" class="uri">https://microbiome.github.io</a>”</p>
<div id="preparing-for-the-course" class="section level2">
<h2><span class="header-section-number">1.1</span> Preparing for the course</h2>
<div id="software-setup" class="section level3">
<h3><span class="header-section-number">1.1.1</span> Software setup</h3>
<p>We recommend using your own laptop. If this is not possible, kindly contact the organizers.</p>
<p>Follow the following steps before the course:</p>
<ol style="list-style-type: decimal">
<li>Install R: Follow the <a href="https://cran.r-project.org">Download link</a> and then choose your operating system. Then in the next step choose R-3.6.1 for Mac or “base” and then R 3.6.1 for Windows. If you already have installed a recent version of R, install Rstudio (see step 2) and then go to step 3. In case installing or loading the packages fails, consider updating R by performing step 1. We tested the material using the newest version (R 3.6.1) as well as R 3.5.</li>
<li>Install/Update Rstudio: Follow the <a href="https://www.rstudio.com/products/rstudio/download/">Download link</a> and choose “Rstudio Desktop” (Open Source License) to download the newest version of Rstudio. Choose the installer that matches your operating system.</li>
<li>Open <a href="code/install_dependencies.R">this Installation script</a> in Rstudio and run the entire script. This will install all required packages for the course.</li>
</ol>
<p>If you are not familiar with R programming, we highly recommend going through <a href="https://datacarpentry.org/R-ecology-lesson/">Data Analysis and Visualization in R for Ecologists</a> by Data Carpentry and further links to R resources in the <a href="https://microbiome.github.io/course_nijmegen2019/study-material.html">study material</a>.</p>
<p>For questions regarding the installation of the software you may contact <a href="mailto:henrikeckermann87@gmail.com">Henrik</a>.</p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>

<a href="schedule.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
