<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>2 Introduction | A Minimal Book Example</title>
  <meta name="description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.12 and GitBook 2.6.7" />

  <meta property="og:title" content="2 Introduction | A Minimal Book Example" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="github-repo" content="microbiome/test" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="2 Introduction | A Minimal Book Example" />
  
  <meta name="twitter:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

<meta name="author" content="Leo Lahti" />


<meta name="date" content="2019-07-29" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html">
<link rel="next" href="citation.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a></li>
<li class="chapter" data-level="2" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>2</b> Introduction</a><ul>
<li class="chapter" data-level="2.1" data-path="introduction.html"><a href="introduction.html#preparing-for-the-course"><i class="fa fa-check"></i><b>2.1</b> Preparing for the course</a></li>
<li class="chapter" data-level="2.2" data-path="introduction.html"><a href="introduction.html#useful-functionsresources"><i class="fa fa-check"></i><b>2.2</b> Useful functions/resources</a></li>
<li class="chapter" data-level="2.3" data-path="introduction.html"><a href="introduction.html#focus"><i class="fa fa-check"></i><b>2.3</b> Focus</a></li>
<li class="chapter" data-level="2.4" data-path="introduction.html"><a href="introduction.html#target-audience"><i class="fa fa-check"></i><b>2.4</b> Target audience</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="citation.html"><a href="citation.html"><i class="fa fa-check"></i><b>3</b> Citation</a><ul>
<li class="chapter" data-level="3.1" data-path="citation.html"><a href="citation.html#license"><i class="fa fa-check"></i><b>3.1</b> License</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html"><i class="fa fa-check"></i><b>4</b> Set-up and Pre-processing</a><ul>
<li class="chapter" data-level="4.1" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#otu-or-asvs-or-sotus"><i class="fa fa-check"></i><b>4.1</b> OTU or ASVs or sOTUs</a></li>
<li class="chapter" data-level="4.2" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#general-overview"><i class="fa fa-check"></i><b>4.2</b> General overview</a></li>
<li class="chapter" data-level="4.3" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#structure"><i class="fa fa-check"></i><b>4.3</b> Structure</a></li>
<li class="chapter" data-level="4.4" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#making-a-phyloseq-object"><i class="fa fa-check"></i><b>4.4</b> Making a phyloseq object</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html"><i class="fa fa-check"></i><b>5</b> Differential abundance testing for univariate data</a><ul>
<li class="chapter" data-level="5.1" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#load-example-data"><i class="fa fa-check"></i><b>5.1</b> Load example data</a></li>
<li class="chapter" data-level="5.2" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#visual-comparison-of-two-groups"><i class="fa fa-check"></i><b>5.2</b> Visual comparison of two groups</a></li>
<li class="chapter" data-level="5.3" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#statistical-comparison-of-two-groups"><i class="fa fa-check"></i><b>5.3</b> Statistical comparison of two groups</a></li>
<li class="chapter" data-level="5.4" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#investigate-assumptions-of-the-t-test"><i class="fa fa-check"></i><b>5.4</b> Investigate assumptions of the t-test</a></li>
<li class="chapter" data-level="5.5" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#compare-results-between-parametric-and-non-parametric-tests"><i class="fa fa-check"></i><b>5.5</b> Compare results between parametric and non-parametric tests</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html"><i class="fa fa-check"></i><b>6</b> Linear models: the role of covariates</a><ul>
<li class="chapter" data-level="6.1" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html#fitting-a-linear-model"><i class="fa fa-check"></i><b>6.1</b> Fitting a linear model</a></li>
<li class="chapter" data-level="6.2" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html#interpreting-linear-model-output"><i class="fa fa-check"></i><b>6.2</b> Interpreting linear model output</a></li>
<li class="chapter" data-level="6.3" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html#covariate-testing"><i class="fa fa-check"></i><b>6.3</b> Covariate testing</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html"><i class="fa fa-check"></i><b>7</b> Advanced models for differential abundance</a><ul>
<li class="chapter" data-level="7.1" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#particular-properties-of-taxonomic-profiling-data"><i class="fa fa-check"></i><b>7.1</b> Particular properties of taxonomic profiling data</a><ul>
<li class="chapter" data-level="7.1.1" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#discrete-count-data"><i class="fa fa-check"></i><b>7.1.1</b> Discrete count data</a></li>
<li class="chapter" data-level="7.1.2" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#sparsity"><i class="fa fa-check"></i><b>7.1.2</b> Sparsity</a></li>
<li class="chapter" data-level="7.1.3" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#rarity"><i class="fa fa-check"></i><b>7.1.3</b> Rarity</a></li>
<li class="chapter" data-level="7.1.4" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#overdispersion"><i class="fa fa-check"></i><b>7.1.4</b> Overdispersion</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#generalized-linear-models-a-brief-overview"><i class="fa fa-check"></i><b>7.2</b> Generalized linear models: a brief overview</a></li>
<li class="chapter" data-level="7.3" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#deseq2-differential-abundance-testing-for-sequencing-data"><i class="fa fa-check"></i><b>7.3</b> DESeq2: differential abundance testing for sequencing data</a><ul>
<li class="chapter" data-level="7.3.1" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#fitting-deseq2"><i class="fa fa-check"></i><b>7.3.1</b> Fitting DESeq2</a></li>
<li class="chapter" data-level="7.3.2" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#comparison-between-deseq2-and-standard-models"><i class="fa fa-check"></i><b>7.3.2</b> Comparison between DESeq2 and standard models</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="multivariate-comparisons-of-microbial-community-composition.html"><a href="multivariate-comparisons-of-microbial-community-composition.html"><i class="fa fa-check"></i><b>8</b> Multivariate comparisons of microbial community composition</a><ul>
<li class="chapter" data-level="8.1" data-path="multivariate-comparisons-of-microbial-community-composition.html"><a href="multivariate-comparisons-of-microbial-community-composition.html#permanova"><i class="fa fa-check"></i><b>8.1</b> PERMANOVA</a></li>
<li class="chapter" data-level="8.2" data-path="multivariate-comparisons-of-microbial-community-composition.html"><a href="multivariate-comparisons-of-microbial-community-composition.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>8.2</b> Checking the homogeneity condition</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="further-exercises.html"><a href="further-exercises.html"><i class="fa fa-check"></i><b>9</b> Further exercises</a><ul>
<li class="chapter" data-level="9.1" data-path="further-exercises.html"><a href="further-exercises.html#compositionality"><i class="fa fa-check"></i><b>9.1</b> Compositionality</a></li>
<li class="chapter" data-level="9.2" data-path="further-exercises.html"><a href="further-exercises.html#redundancy-analysis-rda"><i class="fa fa-check"></i><b>9.2</b> Redundancy analysis (RDA)</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="citation-1.html"><a href="citation-1.html"><i class="fa fa-check"></i><b>10</b> Citation</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">A Minimal Book Example</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="introduction" class="section level1">
<h1><span class="header-section-number">2</span> Introduction</h1>
<center>
<img src="images/logos.png" alt="microbiome R package" />
</center>
<p>Three day spring school on basics of high throughput 16S rRNA gene sequencing data analysis. This spring school is organised in collaboration with <a href="https://www.vlaggraduateschool.nl/en.htm">VLAG graduate school</a>, <a href="https://www.wur.nl/en/Expertise-Services/Chair-groups/Agrotechnology-and-Food-Sciences/Laboratory-of-Microbiology.htm">Laboratory of Microbiology</a>, <a href="https://www.wur.nl/en/Expertise-Services/Chair-groups/Agrotechnology-and-Food-Sciences/Laboratory-of-Systems-and-Synthetic-Biology.htm">Laboratory of Systems and Synthetic Biology</a>, Wageningen University &amp; Research, the Netherlands and <a href="https://www.utu.fi/en/units/sci/units/math/Pages/home.aspx">Department of Mathematics and Statistics University of Turku</a>, Finland. This year it will be held at <a href="https://www.wur.nl/en.htm">Wageningen University &amp; Research</a>. We will cover topics related to basics of sequencing and microbial community analysis. We will use <a href="https://f1000research.com/articles/5-1791/v1">NG-Tax</a> and <a href="https://www.r-project.org/">R</a> based tools like <a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0061217">Phyloseq</a>, <a href="http://microbiome.github.io/microbiome/">microbiome</a> and <a href="http://ggplot2.org/">ggplot2</a> for downstream data analysis and visualization.</p>
<div id="preparing-for-the-course" class="section level2">
<h2><span class="header-section-number">2.1</span> Preparing for the course</h2>
<p>We recommend using your own laptop. If this is not possible, kindly contact the organizers.</p>
<p>Install the following software before the course:</p>
<ul>
<li>R 3.5.1 <a href="https://cran.r-project.org/bin/windows/base/old/3.5.1/">Download link</a>.<br />
</li>
<li>Rstudio <a href="https://www.rstudio.com/products/rstudio/download/">Download link</a>.</li>
</ul>
<p>*Make sure you have admin rights on the machine you install these softwares to avoid access issues.</p>
<p>Download the master branch of the the github repository for the <a href="https://github.com/mibwurrepo/Microbial-bioinformatics-introductory-course-Material-2018">course</a>.</p>
<img src="images/Gihub_repo_step1.png" alt="microbiome R package" />
</center>
<p>Unzip the <code>Microbial-bioinformatics-introductory-course-Material-2018-master.zip</code> folder. Unzip the <code>humanmicrobiome.zip</code> subfolder within the <code>input_data</code> folder so that the <em>humanmicrobiome.biom</em> file is in <code>input_data</code> folder.<br />
open the <code>microbiometutorials.Rproj</code></p>
<p>The script to set-up the RStudio environment with required package installations can be found in the folder named <code>scripts/setup_microbiome_analysis.R</code>. In the image below, there is description of how to run this code.</p>
<img src="images/RStudio_IDE.png" alt="microbiome R package" />
</center>
</div>
<div id="useful-functionsresources" class="section level2">
<h2><span class="header-section-number">2.2</span> Useful functions/resources</h2>
<ul>
<li><a href="https://www.rstudio.com/wp-content/uploads/2016/10/r-cheat-sheet-3.pdf">Base R</a><br />
</li>
<li><a href="https://www.rstudio.com/wp-content/uploads/2016/03/rmarkdown-cheatsheet-2.0.pdf">R Markdown</a><br />
</li>
<li><a href="https://www.rstudio.com/wp-content/uploads/2016/01/rstudio-IDE-cheatsheet.pdf">RStudio IDE</a><br />
</li>
<li><a href="https://www.rstudio.com/wp-content/uploads/2016/11/ggplot2-cheatsheet-2.1.pdf">ggplot2</a><br />
</li>
<li><a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0061217">phyloseq</a><br />
</li>
<li><a href="http://microbiome.github.io/microbiome/">microbiome</a><br />
</li>
<li><a href="http://www.cookbook-r.com/Graphs/">R graphics cookbook</a><br />
</li>
<li><a href="https://microsud.github.io/Tools-Microbiome-Analysis/">List of R tools for microbiome analysis</a></li>
</ul>
</div>
<div id="focus" class="section level2">
<h2><span class="header-section-number">2.3</span> Focus</h2>
<p>The primary aim is introduce microbial community data analysis. There will be talks and discussion on theory and methodology for analysis of microbial community data. We will cover topics ranging from design of studies, sequencing technologies, importance of controls and standardized DNA processing. Supervised hands-on training covering analyses from raw reads using NG-Tax, downstream analysis in R for exploration and analysis of microbiome sequencing data will be priority. There will be a strong focus on using R, R Studio for moving towards reproducible and open science.</p>
</div>
<div id="target-audience" class="section level2">
<h2><span class="header-section-number">2.4</span> Target audience</h2>
<p>Anyone who plans to or is currently doing high throughput microbial community analysis.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="citation.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
