<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>5 Linear models: the role of covariates | Introduction to microbiome data science</title>
  <meta name="description" content="Course material" />
  <meta name="generator" content="bookdown 0.12 and GitBook 2.6.7" />

  <meta property="og:title" content="5 Linear models: the role of covariates | Introduction to microbiome data science" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Course material" />
  <meta name="github-repo" content="microbiome/test" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="5 Linear models: the role of covariates | Introduction to microbiome data science" />
  
  <meta name="twitter:description" content="Course material" />
  

<meta name="author" content="Leo Lahti" />


<meta name="date" content="2019-07-29" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="differential-abundance-testing-for-univariate-data.html">
<link rel="next" href="advanced-models-for-differential-abundance.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#preparing-for-the-course"><i class="fa fa-check"></i><b>1.1</b> Preparing for the course</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#useful-functionsresources"><i class="fa fa-check"></i><b>1.2</b> Useful functions/resources</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#focus"><i class="fa fa-check"></i><b>1.3</b> Focus</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#target-audience"><i class="fa fa-check"></i><b>1.4</b> Target audience</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#citation"><i class="fa fa-check"></i><b>1.5</b> Citation</a></li>
<li class="chapter" data-level="1.6" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i><b>1.6</b> License</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html"><i class="fa fa-check"></i><b>2</b> Set-up and Pre-processing</a><ul>
<li class="chapter" data-level="2.1" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#otu-or-asvs-or-sotus"><i class="fa fa-check"></i><b>2.1</b> OTU or ASVs or sOTUs</a></li>
<li class="chapter" data-level="2.2" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#general-overview"><i class="fa fa-check"></i><b>2.2</b> General overview</a></li>
<li class="chapter" data-level="2.3" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#structure"><i class="fa fa-check"></i><b>2.3</b> Structure</a></li>
<li class="chapter" data-level="2.4" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#making-a-phyloseq-object"><i class="fa fa-check"></i><b>2.4</b> Making a phyloseq object</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="course-overview.html"><a href="course-overview.html"><i class="fa fa-check"></i><b>3</b> Course overview</a><ul>
<li class="chapter" data-level="3.1" data-path="course-overview.html"><a href="course-overview.html#september"><i class="fa fa-check"></i><b>3.1</b> 12 September:</a></li>
<li class="chapter" data-level="3.2" data-path="course-overview.html"><a href="course-overview.html#september-1"><i class="fa fa-check"></i><b>3.2</b> 13 September:</a></li>
<li class="chapter" data-level="3.3" data-path="course-overview.html"><a href="course-overview.html#material"><i class="fa fa-check"></i><b>3.3</b> Material</a></li>
<li class="chapter" data-level="3.4" data-path="course-overview.html"><a href="course-overview.html#literature"><i class="fa fa-check"></i><b>3.4</b> Literature</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html"><i class="fa fa-check"></i><b>4</b> Differential abundance testing for univariate data</a><ul>
<li class="chapter" data-level="4.1" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#load-example-data"><i class="fa fa-check"></i><b>4.1</b> Load example data</a></li>
<li class="chapter" data-level="4.2" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#visual-comparison-of-two-groups"><i class="fa fa-check"></i><b>4.2</b> Visual comparison of two groups</a></li>
<li class="chapter" data-level="4.3" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#statistical-comparison-of-two-groups"><i class="fa fa-check"></i><b>4.3</b> Statistical comparison of two groups</a></li>
<li class="chapter" data-level="4.4" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#investigate-assumptions-of-the-t-test"><i class="fa fa-check"></i><b>4.4</b> Investigate assumptions of the t-test</a></li>
<li class="chapter" data-level="4.5" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#compare-results-between-parametric-and-non-parametric-tests"><i class="fa fa-check"></i><b>4.5</b> Compare results between parametric and non-parametric tests</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html"><i class="fa fa-check"></i><b>5</b> Linear models: the role of covariates</a><ul>
<li class="chapter" data-level="5.1" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html#fitting-a-linear-model"><i class="fa fa-check"></i><b>5.1</b> Fitting a linear model</a></li>
<li class="chapter" data-level="5.2" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html#interpreting-linear-model-output"><i class="fa fa-check"></i><b>5.2</b> Interpreting linear model output</a></li>
<li class="chapter" data-level="5.3" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html#covariate-testing"><i class="fa fa-check"></i><b>5.3</b> Covariate testing</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html"><i class="fa fa-check"></i><b>6</b> Advanced models for differential abundance</a><ul>
<li class="chapter" data-level="6.1" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#particular-properties-of-taxonomic-profiling-data"><i class="fa fa-check"></i><b>6.1</b> Particular properties of taxonomic profiling data</a><ul>
<li class="chapter" data-level="6.1.1" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#discrete-count-data"><i class="fa fa-check"></i><b>6.1.1</b> Discrete count data</a></li>
<li class="chapter" data-level="6.1.2" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#sparsity"><i class="fa fa-check"></i><b>6.1.2</b> Sparsity</a></li>
<li class="chapter" data-level="6.1.3" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#rarity"><i class="fa fa-check"></i><b>6.1.3</b> Rarity</a></li>
<li class="chapter" data-level="6.1.4" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#overdispersion"><i class="fa fa-check"></i><b>6.1.4</b> Overdispersion</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#generalized-linear-models-a-brief-overview"><i class="fa fa-check"></i><b>6.2</b> Generalized linear models: a brief overview</a></li>
<li class="chapter" data-level="6.3" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#deseq2-differential-abundance-testing-for-sequencing-data"><i class="fa fa-check"></i><b>6.3</b> DESeq2: differential abundance testing for sequencing data</a><ul>
<li class="chapter" data-level="6.3.1" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#fitting-deseq2"><i class="fa fa-check"></i><b>6.3.1</b> Fitting DESeq2</a></li>
<li class="chapter" data-level="6.3.2" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#comparison-between-deseq2-and-standard-models"><i class="fa fa-check"></i><b>6.3.2</b> Comparison between DESeq2 and standard models</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="multivariate-comparisons-of-microbial-community-composition.html"><a href="multivariate-comparisons-of-microbial-community-composition.html"><i class="fa fa-check"></i><b>7</b> Multivariate comparisons of microbial community composition</a><ul>
<li class="chapter" data-level="7.1" data-path="multivariate-comparisons-of-microbial-community-composition.html"><a href="multivariate-comparisons-of-microbial-community-composition.html#permanova"><i class="fa fa-check"></i><b>7.1</b> PERMANOVA</a></li>
<li class="chapter" data-level="7.2" data-path="multivariate-comparisons-of-microbial-community-composition.html"><a href="multivariate-comparisons-of-microbial-community-composition.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>7.2</b> Checking the homogeneity condition</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="further-exercises.html"><a href="further-exercises.html"><i class="fa fa-check"></i><b>8</b> Further exercises</a><ul>
<li class="chapter" data-level="8.1" data-path="further-exercises.html"><a href="further-exercises.html#compositionality"><i class="fa fa-check"></i><b>8.1</b> Compositionality</a></li>
<li class="chapter" data-level="8.2" data-path="further-exercises.html"><a href="further-exercises.html#redundancy-analysis-rda"><i class="fa fa-check"></i><b>8.2</b> Redundancy analysis (RDA)</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="citation-1.html"><a href="citation-1.html"><i class="fa fa-check"></i><b>9</b> Citation</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Introduction to microbiome data science</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="linear-models-the-role-of-covariates" class="section level1">
<h1><span class="header-section-number">5</span> Linear models: the role of covariates</h1>
<p>This section provides hands-on introduction to linear (and generalized linear) models.</p>
<p><strong>Task Fit linear model to compare abundance between the two groups.</strong> You can use functions lm or glm, for instance.</p>
<div id="fitting-a-linear-model" class="section level2">
<h2><span class="header-section-number">5.1</span> Fitting a linear model</h2>
<p>Let us compare two groups with a linear model. We use Log10 abundances since this is closer to the Gaussian assumptions than the absolute count data. Fit a linear model with Gaussian variation as follows:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">res &lt;-<span class="st"> </span><span class="kw">glm</span>(Log10_Abundance <span class="op">~</span><span class="st"> </span>Group, <span class="dt">data =</span> df, <span class="dt">family =</span> <span class="st">&quot;gaussian&quot;</span>)</code></pre></div>
</div>
<div id="interpreting-linear-model-output" class="section level2">
<h2><span class="header-section-number">5.2</span> Interpreting linear model output</h2>
<p>Investigate the model coefficients:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">knitr<span class="op">::</span><span class="kw">kable</span>(<span class="kw">summary</span>(res)<span class="op">$</span>coefficients, <span class="dt">digits =</span> <span class="dv">5</span>)</code></pre></div>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">Estimate</th>
<th align="right">Std. Error</th>
<th align="right">t value</th>
<th align="right">Pr(&gt;|t|)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>(Intercept)</td>
<td align="right">0.64825</td>
<td align="right">0.02877</td>
<td align="right">22.53405</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td>GroupAFR</td>
<td align="right">0.20313</td>
<td align="right">0.04308</td>
<td align="right">4.71530</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<p>The intercept equals to the mean in the first group:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">print</span>(<span class="kw">mean</span>(<span class="kw">subset</span>(df, Group <span class="op">==</span><span class="st"> &quot;AAM&quot;</span>)<span class="op">$</span>Log10_Abundance))</code></pre></div>
<pre><code>## [1] 0.6482493</code></pre>
<p>The group term equals to the difference between group means:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">print</span>(<span class="kw">mean</span>(<span class="kw">subset</span>(df, Group <span class="op">==</span><span class="st"> &quot;AFR&quot;</span>)<span class="op">$</span>Log10_Abundance) <span class="op">-</span>
<span class="st">      </span><span class="kw">mean</span>(<span class="kw">subset</span>(df, Group <span class="op">==</span><span class="st"> &quot;AAM&quot;</span>)<span class="op">$</span>Log10_Abundance))</code></pre></div>
<pre><code>## [1] 0.2031287</code></pre>
<p>Note that the linear model (default) significance equals to t-test assuming equal variances.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">print</span>(<span class="kw">t.test</span>(Log10_Abundance <span class="op">~</span><span class="st"> </span>Group, <span class="dt">data =</span> df, <span class="dt">var.equal=</span><span class="ot">TRUE</span>)<span class="op">$</span>p.value)</code></pre></div>
<pre><code>## [1] 4.284318e-06</code></pre>
</div>
<div id="covariate-testing" class="section level2">
<h2><span class="header-section-number">5.3</span> Covariate testing</h2>
<p><strong>Task: Investigate how sex and bmi affect the results. </strong></p>
<p>An important advantage of linear and generalized linear models, compared to plain t-test is that they allow incorporating additional variables, such as potential confounders (age, BMI, gender..):</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Add a covariate:</span>
df<span class="op">$</span>sex &lt;-<span class="st"> </span><span class="kw">meta</span>(d)<span class="op">$</span>sex
df<span class="op">$</span>bmi_group &lt;-<span class="st"> </span><span class="kw">meta</span>(d)<span class="op">$</span>bmi_group

<span class="co"># Fit the model:</span>
res &lt;-<span class="st"> </span><span class="kw">glm</span>(Log10_Abundance <span class="op">~</span><span class="st"> </span>Group <span class="op">+</span><span class="st"> </span>sex <span class="op">+</span><span class="st"> </span>bmi_group, <span class="dt">data =</span> df, <span class="dt">family =</span> <span class="st">&quot;gaussian&quot;</span>)</code></pre></div>
<p>We can even include interaction terms:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">res &lt;-<span class="st"> </span><span class="kw">glm</span>(Log10_Abundance <span class="op">~</span><span class="st"> </span>Group <span class="op">*</span><span class="st"> </span>sex <span class="op">*</span><span class="st"> </span>bmi_group, <span class="dt">data =</span> df, <span class="dt">family =</span> <span class="st">&quot;gaussian&quot;</span>)
<span class="kw">kable</span>(<span class="kw">coefficients</span>(res))</code></pre></div>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>(Intercept)</td>
<td align="right">0.8736029</td>
</tr>
<tr class="even">
<td>GroupAFR</td>
<td align="right">0.0261450</td>
</tr>
<tr class="odd">
<td>sexmale</td>
<td align="right">0.0022871</td>
</tr>
<tr class="even">
<td>bmi_groupoverweight</td>
<td align="right">-0.3160401</td>
</tr>
<tr class="odd">
<td>bmi_groupobese</td>
<td align="right">-0.2117135</td>
</tr>
<tr class="even">
<td>GroupAFR:sexmale</td>
<td align="right">-0.2180267</td>
</tr>
<tr class="odd">
<td>GroupAFR:bmi_groupoverweight</td>
<td align="right">0.5360971</td>
</tr>
<tr class="even">
<td>GroupAFR:bmi_groupobese</td>
<td align="right">0.2396655</td>
</tr>
<tr class="odd">
<td>sexmale:bmi_groupoverweight</td>
<td align="right">0.0133203</td>
</tr>
<tr class="even">
<td>sexmale:bmi_groupobese</td>
<td align="right">NA</td>
</tr>
<tr class="odd">
<td>GroupAFR:sexmale:bmi_groupoverweight</td>
<td align="right">NA</td>
</tr>
<tr class="even">
<td>GroupAFR:sexmale:bmi_groupobese</td>
<td align="right">NA</td>
</tr>
</tbody>
</table>
<p>For more examples on using and analysing linear models, see statmethods <a href="https://www.statmethods.net/stats/regression.html">regression</a> and [ANOVA](See also <a href="https://www.statmethods.net/stats/anova.html">statmethods</a> tutorials. <strong>Try to adapt those examples on our microbiome example data data sets</strong>.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="differential-abundance-testing-for-univariate-data.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="advanced-models-for-differential-abundance.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
